source/util.cpp

```diff
@@ -73,6 +73,23 @@ string replace_(string const &s, string const & from, string const &to)
 	return r;
 }
 
+/// Move all occurances of non-alphanumerical characters to the end of the string and replace them with their ASCII codes.
+std::string to_alphanumerical(const std::string& input)
+{
+    std::string output;
+    std::string suffix;
+    size_t pos=0;
+    for (auto c: input) {
+    
+        if ( ( c>='0' && c<='9') ||  ( c>='a' && c<='z') || ( c>='A' && c<='Z') ||  c=='_' )
+            output+=c;
+        else
+            { suffix+='_'; suffix+=std::to_string(pos); suffix+='_'; suffix+=std::to_string(int(c)); }
+     pos++;
```

Comment:
please do not use use post-increment, while compiler may optimize this technically this is sub-optimal. Please see https://isocpp.org/wiki/faq/operator-overloading#increment-pre-post-speed

Reply:
That is not important for integer types, but yes.